pkg/gui/controllers/helpers/host_helper.go

```diff
@@ -24,18 +24,28 @@ func NewHostHelper(
 }
 
 func (self *HostHelper) GetPullRequestURL(from string, to string) (string, error) {
-	return self.getHostingServiceMgr().GetPullRequestURL(from, to)
+	mgr, err := self.getHostingServiceMgr()
+	if err != nil {
+		return "", err
+	}
+	return mgr.GetPullRequestURL(from, to)
 }
 
 func (self *HostHelper) GetCommitURL(commitSha string) (string, error) {
-	return self.getHostingServiceMgr().GetCommitURL(commitSha)
+	mgr, err := self.getHostingServiceMgr()
+	if err != nil {
+		return "", err
+	}
+	return mgr.GetCommitURL(commitSha)
 }
 
 // getting this on every request rather than storing it in state in case our remoteURL changes
-// from one invocation to the next. Note however that we're currently caching config
-// results so we might want to invalidate the cache here if it becomes a problem.
-func (self *HostHelper) getHostingServiceMgr() *hosting_service.HostingServiceMgr {
-	remoteUrl := self.c.Git().Config.GetRemoteURL()
+// from one invocation to the next.
+func (self *HostHelper) getHostingServiceMgr() (*hosting_service.HostingServiceMgr, error) {
+	remoteUrl, err := self.c.Git().Remote.GetRemoteURL("origin")
```

Comment:
Does this add a dependence on `origin` that wasn't there previously? Like if we had a repo where the main remote wasn't called `origin`, did that already break?

Reply:
I'm not changing the behavior in this regard, as `GetRemoteURL()` is implemented as `self.gitConfig.Get("remote.origin.url")`.

It would be nice to improve this somehow, but I'm actually not sure how, since git doesn't have a clear concept of "the main remote".

Reply:
One way to solve it is to always ask the user what remote their talking about, but it does add more friction so I'm not sure if I want to commit to it.